### ISSUE TITLE
withNuxt is not a function

### ISSUE THREAD

**Issue by edimitchel** [nuxt/eslint#347]
### Environment

------------------------------
- Operating System: Darwin
- Node Version:     v20.11.1
- Nuxt Version:     3.11.1
- CLI Version:      3.11.0
- Nitro Version:    2.9.4
- Package Manager:  npm@10.2.4
- Builder:          -
- User Config:      devtools, modules
- Runtime Modules:  @nuxt/eslint@0.3.0-alpha.0
- Build Modules:    -
------------------------------

### Package

@nuxt/eslint

### Reproduction

https://stackblitz.com/edit/nuxt-starter-in9jep?file=eslint.config.js,nuxt.config.ts,.nuxt%2Feslint.config.mjs

### Describe the bug

I don't know if it's intended but by trying the version alpha 0.3.0, ESLint is complaining about withNuxt is not a function.

By looking in the code from `.nuxt/eslint.config.mjs`, it seems totally right, what is exported from this file is an array, not a function at all.

```
import createConfigForNuxt from '@nuxt/eslint-config/flat';
export default [
...createConfigForNuxt(
```

This is fixable by using `withNuxt` without braces.

### Additional context

Reproductible on local by downloading the stackblitz project.

### Logs

```shell-script
TypeError: withNuxt is not a function
    at ../nuxt-starter-in9jep/eslint.config.js?mtime=1710966190000:4:16
    at ModuleJob.run (node:internal/modules/esm/module_job:194:25)
```

**Comment by edimitchel** [comment]
This issue occurs by using the wrong version of the module (alpha instead of beta).
It works fine on beta channel
